### ISSUE TITLE
Filter meta tags thumbnail sizes based on platform

### ISSUE THREAD

**Issue by roborourke** [humanmade/altis-seo#24]
In addition to setting appropriate image sizes for each platform using tachyon we should filter the default thumbnail size passed down through all contexts to something more reasonable than the full original image size.

- Twitter: 1200x600  (2:1) crop
- Opengraph: 1200x627 (16:9) no crop
- Default: 1200x1200 no crop

To do this use the following filter from the `meta-tags` plugin:

https://github.com/humanmade/meta-tags/blob/master/inc/namespace.php#L255

So for each type you would use the filters:

- `hm.metatags.context.default`
- `hm.metatags.context.opengraph`
- `hm.metatags.context.twitter`

Then check for the existence of `$meta['image]` and do the following:

- check if it's already a tachyon image (string match against the `TACHYON_URL` constant)
  - if yes then use `add_query_arg()` to set the `resize` (crop) or `fit` (no crop) parameter to that listed above
  - if no then use `tachyon_url()` and pass the `resize` or `fit` parameter in the 2nd parameter of that function

Acceptance criteria:

- [x] Filters are not applied if tachyon is disabled in config
- [x] Opengraph meta tag `og:image` contains a tachyon URL set to 1200x627 with no cropping
- [x] Twitter card meta tag `twitter:image` contains a tachyon URL set to 1200x600 with cropping
- [ ] calling `HM\MetaTags\get_meta_for_context()` with no arguments returns an array with a key `image` that is a tachyon URL with constraining dimensions of 1200x1200 with no cropping

To test the above upload a large featured image to a post and check the meta tags output in the `<head>` on the post page.

**Comment by dashaluna** [comment]
@roborourke as far as I can tell we're getting `full` size of the image here https://github.com/humanmade/meta-tags/blob/master/inc/namespace.php#L96

I've followed the function call trail and the only filter I came across is this one:
https://github.com/WordPress/WordPress/blob/master/wp-includes/media.php#L985

Can we use that filter to re-write width and height in the `$image` array? or do we need to re-request the image again to make sure it exists.

Does each class (twitter/opengraph) also needs to register its own thumb image size?

A bit confused here, would appreciate a bit more direction. Thank you.
